invenio_app_ils/records/related/__init__.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Related records."""
+
+from flask import current_app
+from invenio_records_rest.utils import obj_or_import_string
+
+from invenio_app_ils.errors import RelatedRecordError
+
+
+def record_to_relation_dump(record, relation):
+    """Convert a record to a relation dump."""
+    return dict(
+        pid=record[record.pid_field],
+        pid_type=record._pid_type,
+        relation_type=relation.name
+    )
+
+
+def get_node(record, relation):
+    """Get node for a record with a relation."""
+    PIDNodeClass = obj_or_import_string(relation.api)
+    return PIDNodeClass(pid=record.pid, relation_type=relation)
+
+
+class RelatedRecords:
+    """Manage related records."""
+
+    def __init__(self, parent):
+        """Initialize related records proxy.
+
+        :param parent: Parent record
+        """
+        self.parent = parent
+        self.changed_related_records = []
+
+    @property
+    def editions(self):
+        """Get related editions."""
+        return self.get_related(RelatedRecords.edition_relation())
+
+    @property
+    def languages(self):
+        """Get related languages."""
+        return self.get_related(RelatedRecords.language_relation())
+
+    def add_edition(self, record):
+        """Add a new related edition."""
+        self.add(record, RelatedRecords.edition_relation())
+
+    def add_language(self, record):
+        """Add a new related language."""
+        self.add(record, RelatedRecords.language_relation())
+
+    def add(self, record, relation):
+        """Add a new related record."""
+        self._add_record(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def _add_record(self, record, relation, parent_node=None):
+        """Add related record."""
+        if record == self.parent:
+            raise RelatedRecordError("Cannot add itself as a related record.")
+        record_node = get_node(record, relation)
+        if record_node.is_parent or record_node.is_child:
+            raise RelatedRecordError(
+                "Failed to add related record because it already has relations"
```

Comment:
Can you please add more details to **ALL** error message or logs?
I would like to see the type and the PID of the parent and child, and the relation details. If we have an exception, we need to have the info to be able to fix it! :)